ykcapi/src/lib.rs

```diff
@@ -61,4 +66,52 @@ mod c_testing {
         Box::leak(bm);
         ret
     }
+
+    #[no_mangle]
+    pub extern "C" fn __yktrace_start_tracing(kind: usize) -> *mut ThreadTracer {
+        let kind: TracingKind = match kind {
+            SW_TRACING => TracingKind::SoftwareTracing,
+            HW_TRACING => TracingKind::HardwareTracing,
+            _ => panic!(),
```

Comment:
Isn't this turning a compile-time error into a run-time error? If you remove the `_=>panic!()` line, the compiler will refuse to compile this if we add a new `TracingKind` and don't expand the `match` accordingly.

Reply:
I'm not sure what you mean. The match has to be exhaustive, yes.

Reply:
My proposal is that you remove the line `_ => panic!()`.

Reply:
That won't work. The match is over an integer (due to C ABI reasons).

```
72 |         let kind: TracingKind = match kind {
   |                                       ^^^^ pattern `_` not covered
```

Reply:
Aha!